src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -62,8 +63,28 @@ public class FunctionsSyncManager : IFunctionsSyncManager, IDisposable
         private readonly SemaphoreSlim _syncSemaphore = new SemaphoreSlim(1, 1);
         private readonly IAzureBlobStorageProvider _azureBlobStorageProvider;
 
+        //Remove readonly for Integration Testing
+        private IConfiguration _configuration;
+
+        private IExtensionsOptionProvider _extensionsOptionProvider;
+
+        private IConcurrencyOptionProvider _concurrencyOptionProvider;
+
         private BlobClient _hashBlobClient;
 
+        private static HashSet<string> _supportedExtensions = new HashSet<string>
+        {
+            // Filter the extensions that supported for the SyncTrigger
+            "durableTask",
```

Comment:
The hope was that like in my prototype we could avoid having any static lists of extensions. Why is this necessary? I'd expect that ExtensionOptionsProvider only captures extension options types configured via [IWebJobsExtensionBuilder.ConfigureOptions](https://github.com/Azure/azure-webjobs-sdk/blob/3f4ec78be9f43bb041937425ced00b341883aa42/src/Microsoft.Azure.WebJobs.Host/Hosting/WebJobsExtensionBuilderExtensions.cs#L65). Was that not the case?

Reply:
This is a part that we need to discuss. I can remove the checking, however, what I thought was, we can use custom bindings. In this case, if the customer creates extremely big Options class (like EventHubs...) Then we'll got very big payload to the SyncTrigger, so that I thought it might be a good idea to limit supported extensions that need payload. However, if you prefer the dynamic approach rather than worrying about it, or better validation approach, it might be good idea as well.

Reply:
Removed.